baybe/searchspace/core.py

```diff
@@ -153,6 +151,9 @@ def from_product(
                 for c in constraints
                 if isinstance(c, ContinuousLinearInequalityConstraint)
             ],
+            constraints_cardinality=[
```

Comment:
Hey @Scienfitz, as agreed, yes, can do. But the open questions are:
* How do we define its type? Do we keep `tuple[ContinuousCardinalityConstraint, ...]`? Anything else would require setting up another empty intermediate class (with no real purpose) or maintaining a union type somewhere...
* If we keep the original version, do we then still want to include the filtering list-comprehension in the remaining code in this PR or rather do it once we generalize the type when (or rather IF) we ever have another nonlinear constraint?

Reply:
i wouldnt mind the intermediate base class, providing the type for the tuple is a valid purpose (im also not necessarily against a maintained union but id prefer the other way)

dont really get the second question, can you rephrase?

Reply:
If we add the intermediate class, then the second question becomes obsolete. It would imply that we put the intermediate ABC as type inside the tuple, and then filter down to cardinality constraints wherever needed. Can do.

(In the other case, the question would have been what to put as a type in the tuple. And even if we kept `tuple[ContinuousCardinalityConstraint, ...]`, the question would have been if we still need the filtering list comprehension downstream to prevent breaking code because we already anticipate that the type in the tuple would become a union/ABC in the future)

Reply:
ok but I prefer the first one unless other problems surface

Reply:
closing this thread since duplicate

Reply:
thanks